fs_gcsfs/opener.py

```diff
@@ -27,4 +28,12 @@ def open_fs(self, fs_url, parse_result, writeable, create, cwd):  # pylint: disa
         else:
             strict = True
 
-        return GCSFS(bucket_name, root_path=root_path, create=create, strict=strict)
+        client = Client()
```

Comment:
I would not "monkey-patch" attributes after construction. Instead you could parse and then use the parsed output to instantiate the `Client`. I assume `parse_result.params.get(key)` returns None if `key` does not exist

Reply:
What's the disadvantage of doing it like this? It's not monkey-patching: `project` is a public field of `Client`, so `client.project = project` is just a regular setter and an entire valid usage of the class' API

Reply:
Hmmh, is it really. I agree that `project`, etc. are arguments of a public `Client` interface, but I cannot find that they are public fields/attributes/properties of `Client`. 

According to https://googleapis.dev/python/storage/latest/_modules/google/cloud/storage/client.html#Client
they are passed on to the base class `ClientWithProject` which calls init on `Client` and `_ClientProjectMixin` in `google.cloud.core`. 

`_ClientProjectMixin` indeed has a `project` attribute defined in init, but IMHO it hardly qualifies as a public attribute which can be safely set after construction.

The intention of my comment was to just decouple as much as possible from the internals of `Client`.

I might be missing something though.